vc/vc.go

```diff
@@ -0,0 +1,198 @@
+type CreateOption func(*createOptions)
+
+// Contexts can be used to add additional contexts to the Verifiable Credential created by [Create]
+func Contexts(contexts ...string) CreateOption {
+	return func(o *createOptions) {
+		if o.contexts != nil {
+			o.contexts = append(o.contexts, contexts...)
+		} else {
+			o.contexts = contexts
+		}
+	}
+}
+
+// Types can be used to add additional types to the Verifiable Credential created by [Create]
+func Types(types ...string) CreateOption {
+	return func(o *createOptions) {
+		if o.types != nil {
+			o.types = append(o.types, types...)
+		} else {
+			o.types = types
+		}
+	}
+}
+
+// ID can be used to override the default ID generated by [Create]
+func ID(id string) CreateOption {
+	return func(o *createOptions) {
+		o.id = id
+	}
+}
+
+// IssuanceDate can be used to override the default issuance date generated by [Create]
+func IssuanceDate(issuanceDate time.Time) CreateOption {
+	return func(o *createOptions) {
+		o.issuanceDate = issuanceDate
+	}
+}
+
+// ExpirationDate can be used to set the expiration date of the Verifiable Credential created by [Create]
+func ExpirationDate(expirationDate time.Time) CreateOption {
+	return func(o *createOptions) {
+		o.expirationDate = expirationDate
+	}
+}
+
+// Create returns a new Verifiable Credential with the provided claims and options.
+// if no options are provided, the following defaults will be used:
+//   - ID: urn:vc:uuid:<uuid>
+//   - Contexts: ["https://www.w3.org/2018/credentials/v1"]
+//   - Types: ["VerifiableCredential"]
+//   - IssuanceDate: time.Now()
+//
+// # Note
+//
+// Any additional contexts or types provided will be appended to the defaults in order to remain conformant with
+// the W3C Verifiable Credential Data Model specification
+func Create[T CredentialSubject](claims T, opts ...CreateOption) DataModel[T] {
+	o := createOptions{
+		id:           fmt.Sprintf("urn:vc:uuid:%s", uuid.New().String()),
+		contexts:     []string{BaseContext},
+		types:        []string{BaseType},
+		issuanceDate: time.Now(),
+	}
+
+	for _, f := range opts {
+		f(&o)
+	}
+
+	cred := DataModel[T]{
+		Context:           o.contexts,
+		Type:              o.types,
+		ID:                o.id,
+		IssuanceDate:      o.issuanceDate.UTC().Format(time.RFC3339),
+		CredentialSubject: claims,
+	}
+
+	if (o.expirationDate != time.Time{}) {
+		cred.ExpirationDate = o.expirationDate.UTC().Format(time.RFC3339)
+	}
+
+	return cred
+}
+
+// Sign returns a signed JWT conformant with the [vc-jwt] format. sets the provided vc as value of
+// the "vc" claim in the jwt. It returns the signed jwt and an error if the signing fails.
+//
+// [vc-jwt]: https://www.w3.org/TR/vc-data-model/#json-web-token
+func (vc DataModel[T]) Sign(bearerDID did.BearerDID, opts ...jwt.SignOpt) (string, error) {
+	vc.Issuer = bearerDID.URI
+	jwtClaims := jwt.Claims{
+		Issuer:  vc.Issuer,
+		JTI:     vc.ID,
+		Subject: vc.CredentialSubject.GetID(),
```

Comment:
So this would assume single-subject VCs, right? If so, I'd imagine we'll need to add support for N-number of subjects at some point

Reply:
https://github.com/TBD54566975/web5-go/issues/72